src/docfx/build/page/AttributeTransformer.cs

```diff
@@ -38,31 +36,31 @@ object Transform(IEnumerable<DataTypeAttribute> attributes, object value, string
             object TransformXrefSpec(IEnumerable<DataTypeAttribute> attributes, object value, string jsonPath)
             {
                 var attribute = attributes.SingleOrDefault(attr => !(attr is XrefPropertyAttribute));
-                extensionData[jsonPath] = new Lazy<JValue>(() => new JValue(TransformContent(context, errors, attribute, value, file, buildChild)), LazyThreadSafetyMode.PublicationOnly);
+                extensionData[jsonPath] = new Lazy<JValue>(() => new JValue(TransformContent(context, attribute, value, file, buildChild)), LazyThreadSafetyMode.PublicationOnly);
                 return null;
             }
         }
 
-        private static object TransformContent(Context context, List<Error> errors, DataTypeAttribute attribute, object value, Document file, Action<Document> buildChild)
+        private static object TransformContent(Context context, DataTypeAttribute attribute, object value, Document file, Action<Document> buildChild)
         {
             if (attribute is HrefAttribute)
             {
                 var (error, link, _) = context.DependencyResolver.ResolveLink((string)value, file, file, buildChild);
-                errors.AddIfNotNull(error);
+                context.Report.Write(file.ToString(), error);
```

Comment:
I feel like errors should be return values, the fact that `TransformContent` takes `errors` as input also feels wrong. Maybe our dev guide needed a section around error handling.

Reply:
It would be good to update the dev guide. The method of taking `errors` as input is usually not thread-safe and the errors would not be logged if the method is executed on-demand. 
In this case, I just don't want to change the signature of the method, which is coupled with `JsonUtility`.